wagtail_review/models.py

```diff
@@ -413,18 +414,19 @@ def start(self, workflow_state, user=None):
         return super().start(workflow_state, user=user)
 
     def user_can_access_editor(self, page, user):
-        return self.reviewers.filter(internal__pk=user.pk).exists()
+        return self.is_reviewer_for_task(user)
 
     def user_can_lock(self, page, user):
-        return self.reviewers.filter(internal__pk=user.pk).exists()
+        return self.is_reviewer_for_task(user)
 
     def user_can_unlock(self, page, user):
         return False
 
     def get_actions(self, page, user, reviewer=None, **kwargs):
-        if not reviewer:
-            reviewer, created = Reviewer.objects.get_or_create(internal=user)
-        if self.reviewers.filter(pk=reviewer.pk).exists() or user.is_superuser:
+        """Returns the possible actions the user can take. Note that this should
+        be able to be called with a user or a reviewer instance alone (ie where user=None)
+        to account for external reviewers"""
+        if self.is_reviewer_for_task(user, reviewer=reviewer) or user.is_superuser:
```

Comment:
This needs to account for `user` being `None`:
```suggestion
        if self.is_reviewer_for_task(user, reviewer=reviewer) or (user and user.is_superuser):
```

Reply:
Good catch, thanks! Committed now